src/main/java/uk/gov/hmcts/darts/authorisation/component/impl/UserIdentityImpl.java

```diff
@@ -24,43 +31,89 @@ public class UserIdentityImpl implements UserIdentity {
-                emailsAddressesObject = jwt.getClaims().get(PREFERRED_USERNAME);
-            }
-            if (emailsAddressesObject == null) {
-                emailsAddressesObject = jwt.getClaims().get(OID);
-            }
-
-            if (emailsAddressesObject instanceof List<?> emails) {
-                if (emails.size() != 1) {
-                    throw new IllegalStateException(String.format(
-                        "Unexpected number of email addresses: %d",
-                        emails.size()
-                    ));
+            if (principalObject instanceof Jwt jwt) {
+                Object emailsAddressesObject = jwt.getClaims().get(EMAILS);
+                if (emailsAddressesObject == null) {
+                    emailsAddressesObject = jwt.getClaims().get(PREFERRED_USERNAME);
                 }
-                Object emailAddressObject = emails.get(0);
+                if (emailsAddressesObject instanceof List<?> emails) {
+                    if (emails.size() != 1) {
+                        throw new IllegalStateException(String.format(
+                            "Unexpected number of email addresses: %d",
+                            emails.size()
+                        ));
+                    }
+                    Object emailAddressObject = emails.get(0);
 
-                if (emailAddressObject instanceof String emailAddress && StringUtils.isNotBlank(emailAddress)) {
+                    if (emailAddressObject instanceof String emailAddress && StringUtils.isNotBlank(emailAddress)) {
+                        return emailAddress;
+                    }
+                } else if (emailsAddressesObject instanceof String emailAddress && StringUtils.isNotBlank(emailAddress)) {
                     return emailAddress;
                 }
-            } else if (emailsAddressesObject instanceof String emailAddress && StringUtils.isNotBlank(emailAddress)) {
-                return emailAddress;
             }
         }
-
         throw new IllegalStateException("Could not obtain email address from principal");
     }
 
+    private String getGuidFromToken() {
+        if (nonNull(SecurityContextHolder.getContext().getAuthentication())) {
+            Object principalObject = SecurityContextHolder.getContext()
+                .getAuthentication()
+                .getPrincipal();
+
+            Object oid = null;
+            if (principalObject instanceof Jwt jwt) {
+                oid = jwt.getClaims().get(OID);
+            }
+            if (nonNull(oid) && oid instanceof String guid && StringUtils.isNotBlank(guid)) {
+                return guid;
+            }
+        }
+        return null;
+    }
+
     public UserAccountEntity getUserAccount() {
-        return userAccountRepository.findByEmailAddressIgnoreCase(getEmailAddress())
-            .orElseThrow(() -> new DartsApiException(USER_DETAILS_INVALID));
+        UserAccountEntity userAccount = null;
+        String guid = getGuidFromToken();
```

Comment:
The getPrincipal JWT stuff inside the 2 private methods should probably move into this top level method and be passed in

Reply:
Just started changing this and it would still need to be in 2 places as it needs to be in 2 places, it also needs to in userHasGlobalAccess. I don't want to be calling the getUserAccount() and then making another call to get the useraccount for the global  access query